server/src/main/java/org/elasticsearch/search/DefaultSearchContext.java

```diff
@@ -706,6 +706,11 @@ public QuerySearchResult queryResult() {
         return queryResult;
     }
 
+    @Override
+    public NestedDocuments getNestedDocuments() {
+        return new NestedDocuments(mapperService(), bitsetFilterCache()::getBitSetProducer);
```

Comment:
looks like if you do indexService.mapperService() you can go ahead and remove mapperService() from SearchContext.

Reply:
Done! Good spot.